src/plugins/gr-he-common/core/shell.py

```diff
@@ -103,5 +103,35 @@ def _setup(self):
                 if not self.environment[ohostedcons.CoreEnv.TMUX_PROCEED]:
                     raise context.Abort('Aborted by user')
 
+        if (
+            not self.environment[ohostedcons.NetworkEnv.FORCE_IPV4]
+            and not self.environment[ohostedcons.NetworkEnv.FORCE_IPV6]
+        ):
+            self.logger.warning(
+                _('It has been detected that either --4 or --6 '
+                  'options were not used.\n '
+                  'On a dual stack environment, IPv6 will be selected '
+                  'and due to a limitation, hosted-engine will be deployed '
+                  'as IPv6 only.\n '
+                  'Make sure that DNS returns only IPv6 '
+                  'or change Java and imageio address resolution.')
+                # TODO: add a page link for changing Java and imageio config
+            )
+            self.environment[
+                ohostedcons.CoreEnv.FORCE_IP_PROCEED
+            ] = self.dialog.queryString(
```

Comment:
Perhaps ovirt-setup-lib's dialog.queryEnvKey? I might be missing something, but it seems like you ask always (if not passing --4/6), not obeying the answer in the answerfile if provided.